### ISSUE TITLE
Discovery: research how to get programmatic access directly to Azure Active Directory (CTM)

### ISSUE THREAD

**Issue by dianamartin** [cityofaustin/atd-data-tech#12677]
HR Application - Azure Directory problem
1. The CTM is broken / not updating the spreadsheet. (to be able to review correct email addresses)
2. Our script can no longer connect to the G drive. We would need to debug that. 

>if anything, i suggest y'all chat with CTM about getting programmatic access directly to Active Directory so that we can fetch this data ourselves: the G drive solution feels brittle.

_Originally posted by @johnclary in https://github.com/cityofaustin/atd-data-tech/issues/12622#issuecomment-1607619864_

**Comment by dianamartin** [comment]
Pinged John D. (CTM) on 6/26
- Told him about our predicament with unable to access the G drive

>But in azure AD yes everyone has read rights to the directory. So the link to your account

>Is it possible whoever was doing it before left and didn't pass that job on to someone else in their group? They should be able to do it still as far as I know

>I think a ticket to the service desk would be a good place to start and see if they have something already they can share. it looks like Jeffery still works for the city, I have never worked with them directly. So not sure what they do exactly. It should be possible, just have to find the correct group responsible.

**Comment by ChristinaTremel** [comment]
I messaged Jeff H and asked him about the spreadsheet that we currently use for checking Active Directory email addresses. He said it still runs and exports data there, but I followed up with: 
> We've been having some issues using the spreadsheet and the script we have been using can no longer connect to the G drive. We can work on debugging that, but John was recommending that we ask y'all about potentially getting programmatic (read-only) access directly to Active Directory so that we can fetch this data ourselves. Can someone access the Azure AD programmatically? We would be using it to check email addresses against Banner email addresses.

Jeff replied that there is an existing API that pulls Azure Active Directory email addresses. [Taffy Dashboard (austintexas.gov)](https://apps.austintexas.gov/apiRest/) 
Trainemployeemember is the API view. He noted that this has the entire azure directory and we would need to start with a request to the service desk to see if we can get approval from CTM Cyber Ops. 

@johnclary does using this API seem like the better direction to go for checking the email addresses against the banner data?? If so I'll submit a ticket to CTM to see if they will give us access to the Trainemployeemember API.
